From b5c6be9387b37a13ef472e854418938c5b112399 Mon Sep 17 00:00:00 2001
From: Emmanuel Jillela <emmanuel.jillela@intel.com>
Date: Thu, 19 Nov 2015 13:04:03 -0700
Subject: [PATCH]    Title: HW Mailbox init message loss bug fix    Issue:
 <HSD:243901>    IssueType: <BUG>    Description: <Atom HW
 mailbox driver was missing init message and never
 initialized.>    Reviewers: <noamman, ejillela>

---
 arch/x86/hw_mailbox/hw_mbox_drv.c |    2 +-
 include/linux/hw_mbox_defs.h      |    4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/x86/hw_mailbox/hw_mbox_drv.c b/arch/x86/hw_mailbox/hw_mbox_drv.c
index 1450433..1dd01c5 100644
--- a/arch/x86/hw_mailbox/hw_mbox_drv.c
+++ b/arch/x86/hw_mailbox/hw_mbox_drv.c
@@ -2995,7 +2995,7 @@ static int hwMbox_recvOpcodeListenerEntry(void *data)
 
     threadData = data;
 
-    if ( IS_INIT_COMPLETE() || IS_SUPERVISOR() )
+    if ( IS_SUPERVISOR() || IS_INITIALIZED())
     {
         PRINTK_INFO("RECV OPCODE LISTENER#%d kernel thread on CID#%d started running...",
                     threadData->index, threadData->cid);
diff --git a/include/linux/hw_mbox_defs.h b/include/linux/hw_mbox_defs.h
index d368784..69329e7 100644
--- a/include/linux/hw_mbox_defs.h
+++ b/include/linux/hw_mbox_defs.h
@@ -771,10 +771,10 @@ typedef Uint8 __hwMbox_opcodeMessageSizeExceededAllowedSize[HW_MBOX_OPCODE_MINIM
     ENTRY(    4  , HW_MBOX_MASTER_PHY_CPU              , HW_MBOX_MASTER_NP_CPU               , hw_mbox_event            , 136       )    \
     ENTRY(    5  , HW_MBOX_MASTER_MOCA_C               , HW_MBOX_MASTER_NP_CPU               , hw_mbox_event            , 8         )    \
     ENTRY(    6  , HW_MBOX_MASTER_NP_CPU               , HW_MBOX_MASTER_BBU_C                , hw_mbox_opcode           , 8         )    \
-    ENTRY(    7  , HW_MBOX_MASTER_NP_CPU               , HW_MBOX_MASTER_APP_CPU              , hw_mbox_opcode           , 64         )    \
+    ENTRY(    7  , HW_MBOX_MASTER_NP_CPU               , HW_MBOX_MASTER_APP_CPU              , hw_mbox_opcode           , 128       )    \
     ENTRY(    8  , HW_MBOX_MASTER_NP_CPU               , HW_MBOX_MASTER_PHY_CPU              , hw_mbox_opcode           , 8         )    \
     ENTRY(    9  , HW_MBOX_MASTER_NP_CPU               , HW_MBOX_MASTER_MOCA_C               , hw_mbox_opcode           , 8         )    \
-    ENTRY(   10  , HW_MBOX_MASTER_APP_CPU              , HW_MBOX_MASTER_NP_CPU               , hw_mbox_opcode           , 64         )
+    ENTRY(   10  , HW_MBOX_MASTER_APP_CPU              , HW_MBOX_MASTER_NP_CPU               , hw_mbox_opcode           , 128       )
 
 /* static channel Table Expanders: */
 
-- 
1.7.9.5

